client/apps/edit/components/section_image_collection/index.coffee

```diff
@@ -0,0 +1,269 @@
+Artwork = require '../../../../models/artwork.coffee'
+imagesLoaded = require 'imagesloaded'
+Input = React.createFactory require '../section_image_set/input.coffee'
+{ div, section, h1, h2, span, img, header, input, a, button, p, ul, li, strong, nav } = React.DOM
+{ resize } = require '../../../../components/resizer/index.coffee'
+
+module.exports = React.createClass
+
+  getInitialState: ->
+    images: @props.section.get('images') or []
+    layout: @props.section.get('layout') or 'overflow_fillwidth'
+    progress: null
+
+  componentDidMount: ->
+    @$list = $(@refs.images.getDOMNode())
+    @setupAutocomplete()
+    imagesLoaded @$list, =>
+      if @state.images.length > 1
+        @toggleFillwidth()
+      else
+        @$list.animate({opacity: 1}, 500)
+
+  componentWillUnmount: ->
+    @autocomplete.remove()
+
+  componentWillUpdate: ->
+    @props.section.set
+      images: @state.images if @state.images.length > 0
+      layout: @state.layout
+
+  setupAutocomplete: ->
+    $el = $(@refs.autocomplete.getDOMNode())
+    @autocomplete = new Autocomplete
+      url: "#{sd.ARTSY_URL}/api/search?q=%QUERY"
+      el: $el
+      filter: (res) ->
+        vals = []
+        for r in res._embedded.results
+          if r.type == 'Artwork'
+            id = r._links.self.href.substr(r._links.self.href.lastIndexOf('/') + 1)
+            vals.push
+              id: id
+              value: r.title
+              thumbnail: r._links.thumbnail.href
+        return vals
+      templates:
+        suggestion: (data) ->
+          """
+            <div class='esic-suggestion' \
+                 style='background-image: url(#{data.thumbnail})'>
+            </div>
+            #{data.value}
+          """
+      selected: @onSelect
+    _.defer -> $el.focus()
+
+  onSelect: (e, selected) ->
+    new Artwork(id: selected.id).fetch
+      success: (artwork) =>
+        newImages = @state.images.concat [artwork.denormalized()]
+        @setState images: newImages
+        $(@refs.autocomplete.getDOMNode()).val('').focus()
+        @toggleFillwidth() if @state.images.length > 1
+
+  upload: (e) ->
+    gemup e.target.files[0],
+      app: sd.GEMINI_APP
+      key: sd.GEMINI_KEY
+      progress: (percent) =>
+        @setState progress: percent
+      add: (src) =>
+        @$list.animate({opacity: 0}, 500)
+        @setState progress: 0.1
+      done: (src) =>
+        image = new Image()
+        image.src = src
+        image.onload = =>
+          newImages = @state.images.concat [ { url: src, type: 'image' } ]
+          @setState images: newImages
+          imagesLoaded @$list, =>
+            if @state.images.length > 1
+              @toggleFillwidth()
+            else
+              @setState progress: null
+              @$list.animate({opacity: 1}, 500)
+
+  changeLayout: (layout) -> =>
+    @setState layout: layout
+    @props.section.set layout: layout
+    if layout == 'overflow_fillwidth' && @state.images.length > 1
```

Comment:
this should be: 
`if layout is 'overflow_fillwidth' and @state.images.length > 1`